### PR TITLE
Clarify the indexer code

### PR DESCRIPTION
**Pull request by auscompgeek**
I kept getting confused when reading most of this code, so I'm sure others reading this would be at least just as confused.  Hopefully this makes things a lot clearer.

This PR should be reviewed commit by commit.

### CODE REVIEW

File: components/indexer.py [hunk 1]
```diff
@@ -112,15 +117,13 @@ def execute(self) -> None:
             intake_main_motor.set(self.intake_motor_speed)
         else:
             intake_main_motor.set(0)
-        for i, motor in enumerate(
-            self.indexer_motors, len(self.all_motors) - len(self.indexer_motors) - 1
-        ):
+        for i, motor in enumerate(self.indexer_motors, 1):
```
Comment (mlists) [inline components/indexer.py:120]: This line could probably still use a comment explaining how the start value is to account for the offset between indexer_motors and first ball

File: components/indexer.py [hunk 2]
```diff
@@ -130,46 +133,35 @@ def execute(self) -> None:
...
+                    self.shimmy_count += 1
+                    if self.shimmy_count > self.shimmy_ticks:
+                        self.left_shimmy = True
+                        self.shimmy_count = 0
             else:
-                self.intake_right_motor.set(0)
-                self.intake_left_motor.set(0)
-        else:
+                self.intake_left_motor.set(self.shimmy_speed)
+                self.intake_right_motor.set(self.shimmy_speed)
```
Comment (mlists) [inline components/indexer.py:159]: This is a functional rather than stylistic change, not sure if we ever want to run them simultaneously due to jam risks

**Review by mlists — CHANGES_REQUESTED** [review]

File: components/indexer.py (hunk 2 above)
Comment (auscompgeek) [reply components/indexer.py:159]: This has exactly the same behaviour as previously - notice which if statement this is for.

File: components/indexer.py (hunk 1 above)
Comment (auscompgeek) [reply components/indexer.py:119]: Done, PTAL.

File: components/indexer.py (hunk 2 above)
Comment (mlists) [reply components/indexer.py:158]: I see, still not sure we want to do this but it is the only way the intake functions without shimmying.

**Review by mlists — APPROVED** [review]

**Pull request merged**